scipy/optimize/_minimize_constrained.py

```diff
@@ -0,0 +1,485 @@
+                   for sparse matrices.
+                - 'AugmentedSystem': The operators
+                   will be computed using the
+                   so-called augmented system approach
+                   explained in [1]_. Exclusive
+                   for sparse matrices.
+                - 'QRFactorization': Compute projections
+                   using QR factorization. Exclusive for
+                   dense matrices.
+                - 'SVDFactorization': Compute projections
+                   using SVD factorization. Exclusive for
+                   dense matrices.
+
+                The factorization methods 'NormalEquation' and
+                'AugmentedSystem' should be used when
+                ``sparse_jacobian=True`` and, on the other
+                hand, the methods 'QRFactorization' and
+                'SVDFactorization' when ``sparse_jacobian=False``.
+
+    callback : callable, optional
+        Called after each iteration:
+
+            callback(OptimizeResult state) -> bool
+
+        If callback returns true the algorithm execution is terminated.
+    max_iter : int, optional
+        Maximum number of algorithm iterations.
+    verbose : {0, 1, 2}, optional
+        Level of algorithm's verbosity:
+
+            * 0 (default) : work silently.
+            * 1 : display a termination report.
+            * 2 : display progress during iterations.
+
+    Returns
+    -------
+    `OptimizeResult` with the following fields defined:
+    x : ndarray, shape (n,)
+        Solution found.
+    s : ndarray, shape (n_ineq,)
+        Slack variables at the solution. ``n_ineq`` is the total number
+        of inequality constraints.
+    v : ndarray, shape (n_ineq + n_eq,)
+        Slack variables at the solution. ``n_ineq`` and ``n_eq`` are the
+        total number of equality and inequality constraints.
+    niter : int
+        Total number of iterations.
+    nfev : int
+        Total number of objective function evaluations.
+    ngev : int
+        Total number of objective function gradient evaluations.
+    nhev : int
+        Total number of lagragian Hessian evaluations.
+    ncev : int
+        Total number of constraint evaluations.
+    njev : int
+        Total number of constraint Jacobian matrix evaluations.
+    niter : int
+        Total number of CG iterations.
+    cg_info : Dict
+        Dictionary containing information about the latest CG iteration:
+
+            - niter : Number of iterations.
+            - stop_cond : Reason for CG subproblem termination:
+
+                1. Iteration limit was reached;
+                2. Reached the trust-region boundary;
+                3. Negative curvature detected;
+                4. Tolerance was satisfied.
+
+            - hits_boundary : True if the proposed step is on the boundary
+              of the trust region.
+
+    execution_time : float
+        Total execution time.
+    trust_radius : float
```

Comment:
Matt, I think this may be a little too much. Someone reading the reference should not have no difficulty understanding what is ``trust_radius``, ``penalty``, ``tolerance`` and ``barrier_parameter ``